src/PermissionRegistrar.php

```diff
@@ -16,29 +18,50 @@ class PermissionRegistrar
     /** @var \Illuminate\Contracts\Cache\Repository */
     protected $cache;
 
-    /** @var string */
-    protected $cacheKey;
-
     /** @var string */
     protected $permissionClass;
 
     /** @var string */
     protected $roleClass;
 
+    /** @var int */
+    public static $cacheExpirationTime;
+
+    /** @var string */
+    public static $cacheKey;
+
+    /** @var string */
+    public static $cacheModelKey;
+
     /** @var bool */
     public static $cacheIsTaggable = false;
 
+    /**
+     * PermissionRegistrar constructor.
+     *
+     * @param \Illuminate\Contracts\Auth\Access\Gate $gate
+     * @param \Illuminate\Contracts\Cache\Repository $cache
+     */
     public function __construct(Gate $gate, Repository $cache)
     {
         $this->gate = $gate;
-        $this->cache = $cache;
         $this->permissionClass = config('permission.models.permission');
         $this->roleClass = config('permission.models.role');
-        $this->cacheKey = config('permission.cache.key');
 
+        self::$cacheExpirationTime = config('permission.cache.expiration_time',
+            config('permission.cache.expiration_time'));
```

Comment:
The second one should be `cache_expiration_time` with underscore, right?

Reply:
Good spot, thank you.